src/promptflow/promptflow/_sdk/entities/_connection.py

```diff
@@ -631,13 +656,101 @@ class CustomConnection(_Connection):
+                }
+            )
         super().__init__(secrets=secrets, configs=configs, **kwargs)
+        self.custom_type = kwargs.get(CustomStrongTypeConnectionConfigs.TYPE, None)
 
     @classmethod
-    def _get_schema_cls(cls):
+    def _get_custom_keys(cls, data):
+        # The data could be either from yaml or from DB.
+        # If from yaml, 'custom_type' and 'module' are outside the configs of data.
+        # If from DB, 'custom_type' and 'module' are within the configs of data.
+        if not data.get(CustomStrongTypeConnectionConfigs.TYPE) or not data.get(
+            CustomStrongTypeConnectionConfigs.MODULE
+        ):
+            if (
+                not data["configs"][CustomStrongTypeConnectionConfigs.FULL_TYPE]
+                or not data["configs"][CustomStrongTypeConnectionConfigs.FULL_MODULE]
+            ):
+                raise ValueError("custom_type and module are required for custom strong type connections.")
+            else:
+                m = data["configs"][CustomStrongTypeConnectionConfigs.FULL_MODULE]
+                custom_cls = data["configs"][CustomStrongTypeConnectionConfigs.FULL_TYPE]
+        else:
+            m = data[CustomStrongTypeConnectionConfigs.MODULE]
+            custom_cls = data[CustomStrongTypeConnectionConfigs.TYPE]
+
+        try:
+            module = importlib.import_module(m)
+            cls = getattr(module, custom_cls)
+        except ImportError:
+            raise ValueError(
+                f"Can't find module {m} in current environment. Please check the module is correctly configured."
+            )
+        except AttributeError:
+            raise ValueError(
+                f"Can't find class {custom_cls} in module {m}. Please check the custom_type is correctly configured."
+            )
+
+        schema_configs = {}
+        schema_secrets = {}
+
+        for k, v in cls.__annotations__.items():
+            if v == Secret:
+                schema_secrets[k] = v
+            else:
+                schema_configs[k] = v
+
+        return schema_configs, schema_secrets
+
+    @classmethod
+    def _get_schema_cls(cls, is_custom_strong_type=False):
+        if is_custom_strong_type:
+            return CustomStrongTypeConnectionSchema
         return CustomConnectionSchema
 
+    @classmethod
+    def _load_from_dict(cls, data: Dict, context: Dict, additional_message: str = None, **kwargs):
+        # Note here are two cases:
+        # 1. When create/load connection data from yaml, the custom_type and module are outside the configs of data.
+        # 2. When update/load connection data from DB, the custom_type and module are within the configs of data.
+        is_custom_strong_type = data.get(CustomStrongTypeConnectionConfigs.TYPE) or (
+            data.get("configs") and data.get("configs").get(CustomStrongTypeConnectionConfigs.FULL_TYPE)
+        )
+        schema_cls = cls._get_schema_cls(is_custom_strong_type=is_custom_strong_type)
```

Comment:
Better to change to:
```
if is_custom_strong_type:
     return CustomStrongTypeConnection._load_from_dict()
```
And move custom strong type related logic in to CustomStrongTypeConnection class, remove the `is_custom_strong_type` in _get_schema_cls

Reply:
Good suggestion! Moved related logic to CustomStrongTypeConnection class